### PR TITLE
Feature/search bar

### PR DESCRIPTION
**Pull request by verena-ifx**
By creating this pull request you agree to the terms in CONTRIBUTING.md.
https://github.com/Infineon/.github/blob/master/CONTRIBUTING.md
--- DO NOT DELETE ANYTHING ABOVE THIS LINE ---

CONTRIBUTING.md also tells you what to expect in the PR process.

Description
Search-bar component added

<details>
  <summary>📦 Published PR as canary version: <code>18.0.3--canary.80.03bbb7ac8b9cc940ef6170547fc3e9f9300183c2.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @infineon/infineon-design-system-stencil@18.0.3--canary.80.03bbb7ac8b9cc940ef6170547fc3e9f9300183c2.0
  # or 
  yarn add @infineon/infineon-design-system-stencil@18.0.3--canary.80.03bbb7ac8b9cc940ef6170547fc3e9f9300183c2.0
  ```
</details>

### CODE REVIEW

File: src/components/search-input/search-input.scss [hunk 1]
```diff
@@ -1,21 +1,68 @@
...
+        color: #8D8786;
+        border: none;
+        width: 100%;
+
+        &:focus {
+          outline: none;
+        }
+      }
+    }
+
```
Comment (tishoyanchev) [inline src/components/search-input/search-input.scss:65]: I am not sure which part of this SCSS, but something is making the `search-input` larger than it should be when nested inside the `ifx-dropdown` component. The width of the `search-input` inside the search variant of the `dropdown` is `262px`, but in Figma (and the way it was before) is `200px`. 
I was not able to immediately understand where does the extra `width` come from.

**Review by tishoyanchev — CHANGES_REQUESTED** [review]

File: src/components/search-bar/search-bar.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  tag: 'ifx-search-bar',
+  styleUrl: 'search-bar.scss',
+  shadow: true,
+})
+export class SearchBar {
+  private inputElement: HTMLInputElement;
+  private debounceSearch: any;
+
+  @Prop() icon: string;
+  @Prop() width: string = '100%';
```
[the remark this answers is not in the document]
Comment (verena-ifx) [reply src/components/search-bar/search-bar.tsx:14]: There are 5 sizes, thats why I added the percentages. I am not sure about the responsiveness tbh. Something that was not 100% clear from the Figma to me

File: src/components/search-bar/search-bar.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            <ifx-search-input style={{ width: this.width }}></ifx-search-input>
+            {this.showCloseButton ? (
+              <ifx-button variant="outline-text" icon="" position="left" href="" target="_blank" color="primary" size="m" onClick={this.handleClick}>
+                Close
+              </ifx-button>
+            ) : null}
+          </div>
+        ) : (
+          <div class="search-bar__icon-wrapper">
+            <ifx-icon icon="search-16" onClick={this.handleClick}></ifx-icon>
```
[the remark this answers is not in the document]
Comment (verena-ifx) [reply src/components/search-bar/search-bar.tsx:50]: good point, havent seen that one

File: src/components/search-bar/search-bar.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import { Component, EventEmitter, h, Event, Prop, State } from '@stencil/core';
+import { debounce } from 'lodash';
+
+@Component({
+  tag: 'ifx-search-bar',
+  styleUrl: 'search-bar.scss',
+  shadow: true,
+})
+export class SearchBar {
```
[the remark this answers is not in the document]
Comment (verena-ifx) [reply src/components/search-bar/search-bar.tsx:9]: Search Input is a separate Component that should be used inside of the Search bar. That is the information I received. I do not see any mobile specifications on figma though.

**Pull request merged**